src/App.js

```diff
@@ -1,19 +1,26 @@
-import React from 'react';
+import React, { useState } from 'react';
+import UserContext from "./contexts/UserContext";
 import GlobalStyle from "./styles/global";
 import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
 import Timeline from './routes/private/Timeline';
 import SignUpRoute from './routes/public/SingUpRoute/SignUpRoute';
 import LogInRoute from './routes/public/LoginRoute/LoginRoute';
 
 function App() {
+
+  const [user, setUser] = useState("");
+
   return (
     <Router>
-      <GlobalStyle />
-      <Switch>
-        <Route exact path="/" component={LogInRoute} />
-        <Route exact path="/sign-up" component={SignUpRoute} />
-        <Route exact path='/timeline' component={Timeline} />
-      </Switch>
+      <UserContext.Provider value={{ user, setUser }}>
+        <GlobalStyle />
+        <Switch>
+          <Route exact path="/sign-up" component={SignUpRoute} />
+          <Route exact path='/timeline' component={Timeline} />
+          <Route exact path="/" component={LogInRoute} />
+          <Route exact path="*" component={LogInRoute} />
```

Comment:
aqui a lógica para redirecionar o usuário para o login caso tente acessar uma rota que não existe

Reply:
opa não foi o código, ao invés de * você pode fazer: 
`<Redirect to="/" />`

testa alterar a linha 21 pelo código acima ;)